### DISCUSSION TITLE
Use a custom rule within the custom rule project itself

### DISCUSSION THREAD

**Discussion by mirobo** [eslint/eslint#17383] [Q&A] [answered]
I have a custom ES lint rule project written in TS and the rule is published as a plugin to a repository.

Now I'd like to have test project **within** the "custom rule"-project that can use the plugin or the rules **without needing to publish the artifact**. In the plugins section I cannot add a directory and I'm not sure whether eslint can digest TypeScript rules directly or if I need to do "tsc --build" and include the generated rules in another way.

Of course there are tests to figure out the rules but in the end I'd like to see how the for example the ES Lint Rule text is rendered in IntelliJ and probably also how fast the rules are applied. For fiddling around with rules it seems like a good option..

Any idea?

**Comment by nzakas** [comment, marked as answer]
ESLint custom rules must be in JavaScript, not TypeScript. If you have a custom rule written in TypeScript in your project, then you'll need to compile it into a JavaScript file for ESLint to be able to load it. 

We have details on how to load a file locally in the documentation:
https://eslint.org/docs/latest/extend/custom-rule-tutorial#step-8-use-the-plugin-locally

**Reply by mirobo** [reply]
Thank you very much, I'll try it out :-) ❤
